Source/AwsCommonRuntimeKit/auth/signing/SigV4HttpRequestSigner.swift

```diff
@@ -31,58 +47,79 @@ public class SigV4HttpRequestSigner {
     ///    - `config`: The `SigningConfig` to use when signing.
     /// - `Throws`: An error of type `AwsCommonRuntimeError` which will pull last error found in the CRT
     /// - `Returns`: Returns a signed http request `HttpRequest`
-    public func signRequest(request: HttpRequest, config: SigningConfig) async throws -> HttpRequest {
-        typealias SignedContinuation = CheckedContinuation<HttpRequest, Error>
+    public static func signRequest(request: HttpRequest, config: SigningConfig, allocator: Allocator = defaultAllocator) async throws -> HttpRequest {
```

Comment:
Is the `HttpRequest` that gets returned, the same `HttpRequest` that was passed in? If so, that's confusing

Seems better to do either:
1) Mutate the request that gets passed in, do not return a value
OR
2) Do not mutate the request that gets passed in, return a new HttpRequest

In theory, I like 2. But in reality, we should probably do 1, since there's lots of complexity over how to do ref-counting on the body stream when we make a copy of the request

Reply:
Thanks, I agree that doing both is a bit weird. I have updated it to approach 2.

Reply:
I just realized that the request signing is supposed to be async. I am not really sure about approach 1. Looking more into it.

Reply:
I have added a comment as discussed after failing to find a better approach for now.